src/shared/sleep-config.c

```diff
@@ -403,14 +387,14 @@ static bool can_s2h(void) {
         return true;
 }
 
-static int can_sleep_internal(const char *verb, bool check_allowed) {
+static int can_sleep_internal(const char *verb, bool check_allowed, SleepConfig *sleep_config) {
         bool allow;
-        _cleanup_strv_free_ char **modes = NULL, **states = NULL;
```

Comment:
The `_cleanup_strv_free_` is no longer needed as these variables will be assigned to pointers from the appropriate `sleep_config` fields in the `sleep_settings` function. Declaring with `_cleanup_strv_free` causes duplicate attempts to free the memory and I didn't think it made sense to `modes` or `states` values.